packages/composites/implied-price-test/src/endpoint/computedPrice.ts

```
@@ -0,0 +1,123 @@
+      description: 'The minimum number of answers needed to return a value for the operand1',
+      default: 1,
+    },
+    operand2Sources: {
+      required: true,
+      type: 'string',
+      array: true,
+      aliases: ['divisorSources'],
+      description: 'An array of source adapters to query for the operand2 value',
+    },
+    operand2Input: {
+      required: true,
+      type: 'string',
+      aliases: ['divisorInput'],
+      description: 'The JSON payload to send to the operand2 sources',
+    },
+    operand2MinAnswers: {
+      required: false,
+      type: 'number',
+      aliases: ['divisorMinAnswers'],
+      description: 'The minimum number of answers needed to return a value for the operand2',
+      default: 1,
+    },
+    operation: {
+      default: 'divide',
+      type: 'string',
+      description: 'The operation to perform on the operands',
+      options: ['divide', 'multiply'],
+    },
+  },
+  [
+    {
+      operand1Sources: ['coingecko'],
+      operand1MinAnswers: 1,
+      operand1Input: '{"from":"LINK","to":"USD","overrides":{"coingecko":{"LINK":"chainlink"}}}',
+      operand2Sources: ['coingecko'],
+      operand2MinAnswers: 1,
+      operand2Input: '{"from":"ETH","to":"USD","overrides":{"coingecko":{"ETH":"ethereum"}}}',
+      operation: 'multiply',
+    },
+  ],
+)
+
+export type BaseEndpointTypes = {
+  Parameters: typeof inputParameters.definition
+  Response: {
+    Result: string
+    Data: {
+      result: string
+    }
+  }
+  Settings: typeof config.settings
+}
+
+export const endpoint = new AdapterEndpoint({
+  name: 'computedPrice',
+  aliases: ['impliedPrice'],
+  transport: computedPriceTransport,
+  inputParameters,
+  customInputValidation: (req): AdapterInputError | undefined => {
+    const { operand1Sources, operand2Sources, operand1MinAnswers, operand2MinAnswers } =
+      req.requestContext.data
+    let e = validateSources(operand1Sources, operand1MinAnswers)
+    if (e) return e
```

Many customInputValidation just throw instead of returning an error.
[The effect is the same](https://github.com/smartcontractkit/ea-framework-js/blob/d052a5d168498b6908b05f51f8e727c2f15199e5/src/validation/index.ts#L86) and it makes the code cleaner.